### ISSUE TITLE
[typescript-resolvers] Create a type for directive args

### ISSUE THREAD

**Issue by victor-accarini** [dotansimha/graphql-code-generator#3418]
I wanted to enforce a better type check when implementing Apollo custom directives, but since the directive arguments are exported inside the directive resolver type I need to create a workaround to retrieve the arguments types.

Since I have several custom directives it would be nice if the type was exported from the generated file.

Now when exporting a Directive we have:
```ts
export type MyDirectiveDirectiveResolver<
  Result,
  Parent,
  ContextType = any,
  Args = { arg1: Maybe<Scalars['String']> }
> = DirectiveResolverFn<Result, Parent, ContextType, Args>;
```

I would like to change it to:
```ts
export type MyDirectiveDirectiveArgs = {
  arg1: Maybe<Scalars['String']>
}
export type MyDirectiveDirectiveResolver<
  Result,
  Parent,
  ContextType = any,
  Args = MyDirectiveDirectiveArgs
> = DirectiveResolverFn<Result, Parent, ContextType, Args>;
```

This way I can import the `Args` type and set in my SchemaDirectiveVisitor for better type checking.

The workaround that I'm using is this:
```ts
/** Gets the third argument type from the function:
export type DirectiveResolverFn<
  TResult = {},
  TParent = {},
  TContext = {},
  TArgs = {}
> = (
  next: NextResolverFn<TResult>,
  parent: TParent,
  args: TArgs,
  context: TContext,
  info: GraphQLResolveInfo,
) => TResult | Promise<TResult>;
*/
type MyDirectiveArgs = Parameters<
  MyDirectiveDirectiveResolver<unknown, unknown, unknown>
>[2];

export class MyDirective extends SchemaDirectiveVisitor {
  args: MyDirectiveArgs;
```

**Comment by ardatan** [comment]
You can try the following canary version with this feature until it gets released;
`1.12.2-alpha-3cb57f00.14`

**Comment by dotansimha** [comment]
Fixed in `1.12.2`! Thanks @ardatan

**Comment by victor-accarini** [comment]
Thanks! @ardatan @dotansimha
